drivers/hwinfo/hwinfo_mcux_syscon.c

```diff
@@ -10,9 +10,14 @@
 #include <string.h>
 #include <zephyr/sys/byteorder.h>
 
+struct uid {
+	uint32_t id[4];
+};
+
 ssize_t z_impl_hwinfo_get_device_id(uint8_t *buffer, size_t length)
 {
-	uint8_t *uid_addr = (uint8_t *) DT_INST_REG_ADDR(0);
+	uint32_t *uid_addr = (uint32_t *) DT_INST_REG_ADDR(0);
```

Comment:
Needs to be volatile if you want to prevent optimizations and unaligned access.

Reply:
Thanks for the reminder -- updated and pushed new commit.